main_server.py

```
@@ -2041,6 +3048,935 @@ def search_files_recursive(directory, target_ext, result_list):
+        if not path:
+            return JSONResponse(content={"exists": False}, status_code=400)
+        
+        exists = os.path.exists(path) and os.path.isfile(path)
+        return JSONResponse(content={"exists": exists})
+        
+    except Exception as e:
+        logger.error(f"检查文件存在失败: {e}")
+        return JSONResponse(content={"exists": False}, status_code=500)
+
+@app.get('/api/find-first-image')
+async def find_first_image(folder: str = None):
```

Ruff (0.14.5)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)